ocaml/tests/test_repository_helpers.ml

```diff
@@ -741,7 +741,11 @@ module UpdateInfoMetaDataOfXml = Generic.MakeStateless (struct
   end
 
   let transform input =
-    try Ok (UpdateInfoMetaData.of_xml (Xml.parse_string input))
+    try
+      Ok
+        (UpdateInfoMetaData.of_xml (Xml.parse_string input) "updateinfo"
+        |> UpdateInfoMetaData.assert_valid
+        )
```

Comment:
I think the validation should happens in the conversion function, as well as picking the field, simplifying this code into: 

```suggestion
      Ok (UpdateMetaData.of_xml (Xml.parse_string input))
```
This means `UpdateInfoMetadata.of_xml` could be defined as
```ocaml
  let of_xml xml =
    let update_info = Repometadata.of_xml xml "updateinfo" in
    UpdateInfoMetaData.assert_valid update_info ;
    update_info
```
This would make this function much easier to use and less error-prone